src/datapilot/utils/utils.py

```diff
@@ -58,3 +63,79 @@ def is_superset_path(superset_path: str, path: str):
         return True
     except ValueError:
         return False
+
+
+def get_changed_files(include_untracked=True):
+    command = ["git", "status", "--porcelain"]
+    if include_untracked:
+        command.append("-uall")
+    result = subprocess.run(command, capture_output=True, text=True)  # noqa
+    changed_files = []
+    for line in result.stdout.splitlines():
+        if line.startswith("??") and include_untracked:
+            changed_files.append(line.split()[1])
+        elif line.startswith(("M", "A", "D", "R", " M", " A", " D", " R")):
+            changed_files.append(line.split()[1])
+    return changed_files
+
+
+def get_tmp_dir_path():
+    tmp_dir = Path(tempfile.gettempdir()) / str(uuid.uuid4())
+    tmp_dir.mkdir(parents=True, exist_ok=True)
+    return tmp_dir
+
+
+class DBTNode(BaseModel):
+    unique_id: str
+    name: str
+    resource_type: str
+    table: str = ""
+
+
+def generate_partial_manifest_catalog(changed_files, manifest_path: str, catalog_path: str):
+    models = [Path(f).stem for f in changed_files]
+
+    subprocess.run(["dbt", "compile", "--models", " ".join(models)])  # noqa
```

Comment:
do you need this? compile also maybe slow for a lot of models.  `dbt parse might be better`